Source/Cake.SendGrid/Email/SendGridEmailProvider.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using Cake.Core;
+using Cake.Core.Annotations;
+
+namespace Cake.SendGrid.Email
+{
+    /// <summary>
+    /// Contains functionality related to SendGrid API
+    /// </summary>
+    [CakeAliasCategory("SendGrid")]
+    public sealed class SendGridEmailProvider
+    {
+        private readonly ICakeContext _context;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SendGridEmailProvider"/> class.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        public SendGridEmailProvider(ICakeContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        /// Send an email through SendGrid
+        /// </summary>
+        /// <param name="senderName">The name of the person sending the email</param>
+        /// <param name="senderAddress">The email address of the person sending the email</param>
+        /// <param name="recipientName">The name of the person who will receive the email</param>
+        /// <param name="recipientAddress">The email address of the person who will recieve the email</param>
+        /// <param name="subject">The subject line of the email</param>
+        /// <param name="htmlContent">The HTML content</param>
+        /// <param name="textContent">The text content</param>
+        /// <param name="settings">The settings to be used when sending the email</param>
+        /// <returns>An instance of <see cref="SendGridEmailResult"/> indicating success/failure</returns>
```

Comment:
It would be great if we would have an example of the usage here in the XML summary as well, similar to what has been done here:

https://github.com/cake-contrib/Cake.Gem/blob/develop/Source/Cake.Gem/GemAliases.cs#L29

It will likely be the same content as you have put on the index.md, but it means that it will show up as an example on the cakebuild.net site:

http://cakebuild.net/api/cake.gem/4f3488d3/56b306a7